code/game/objects/structures/tables_racks.dm

```diff
@@ -154,6 +292,41 @@
 				user.stop_pulling()
 	return ..()
 
+/obj/structure/table/attack_hand_secondary(mob/user, list/modifiers)
+	. = ..()
+	if(!istype(user) || !user.can_interact_with(src))
+		return FALSE
+
+	if(!can_flip)
+		return
+
+	if(!is_flipped)
+		user.balloon_alert_to_viewers("flipping table...")
+		if(!do_after(user, max_integrity * 0.25))
+			return
+
+		flip_table(get_dir(user, src))
+
+		return
+
+	else
+		user.balloon_alert_to_viewers("flipping table upright...")
+		if(do_after(user, max_integrity * 0.25, src))
+			unflip_table()
+			return TRUE
+
+/obj/structure/table/proc/is_able_to_throw(obj/structure/table, atom/movable/movable_entity)
+	if (movable_entity == table) //Thing is not the table
+		return FALSE
+	if (movable_entity.anchored) //Thing isn't anchored
+		return FALSE
+	if(!isliving(movable_entity) && !isobj(movable_entity)) //Thing isn't an obj or mob
+		return FALSE
```

Comment:
Since you are not supporting throwing living mobs off the table, you should restrict table flipping in `attack_hand_secondary` if a mob is standing on it.